### ISSUE TITLE
segfault with clitkDicom2Image

### ISSUE THREAD

**Issue by djboersma** [open-vv/vv#70]
A typical dicom directory in radiotherapy has CT, RD, RS and RP dicom files in it. When we try to convert the CT files into an MHD file, we get a segfault:

```
clitkDicom2Image -v -o /tmp/ct.mhd /path/to/CT*.dcm
Using GDCM-2.x
Reading </path/to/CT... [200 times]
Segmentation fault (core dumped)
```

There are 200 dicom files for this CT image. It does not matter whether we add a relative or an absolute path to the file pattern, we always get the same crash. I am using the git version cloned on 2019-01-05, built with VTK 7.1.1 and ITK 4.13.0.

After some lengthy debugging, I discovered that the `sliceIndex` vector, which is defined in https://github.com/open-vv/vv/blob/master/tools/clitkDicom2Image.cxx#L162 and is supposed to be filled in the next few lines, remains empty, so that either in line 181 or in 200, when `sliceIndex[0]` is accessed without any prior length check, a segfault occurs.

The `sliceIndex` vector remains empty because the list of file names generated by the "file names generator" (line 165: https://github.com/open-vv/vv/blob/master/tools/clitkDicom2Image.cxx#L165) actually contains only one file, one that is not a CT file at all, but one of the dose files (RD).

When I run with the `-s` option (use series UID), then we get a fatal error saying: `GDCMSeriesFileNames (0xb150fe0): No Series were found`.

When we ran this tool with an older version of VV (git develop from maybe half a year ago) but the same ITK version (4.13.0) on the same image set, it worked like it is supposed to work, namely produce an MHD file of the CT dicom image. We also tried recompiling with 4.12.2, but that does not help much: the `-s` option now only results in a warning about the missing series (instead of a fatal error), and then it runs into the segfault like before.

**Comment by djboersma** [comment]
I see that there were several changes in this code during December. I checked out a version from before those changes (December 4, 2018, commit be384dae0a3d42503ea8af4d471c455943273822 by dsarrut) and with that version, we can run clitkDicom2Image on our DICOM CT data without getting a segfault.

**Comment by tbaudier** [comment]
Hi David,

Thank you to report the bug. In fact I did a previous modifications in December to deal a problem reported on the mailing list. I did modifications with the latest 3 commits: https://github.com/open-vv/vv/commit/d647919fe732d73ed24138daaec16080c00df892, https://github.com/open-vv/vv/commit/cd904bc3cb27133a8a556c13c21a37d6415a6cc8, https://github.com/open-vv/vv/commit/48727138e88f056b67312603560e600362346462.
In my case, it's seems to be OK. And you?

Thank you
Thomas
